src/Geometry.cpp

```diff
@@ -1011,6 +1011,114 @@ void Geometry::computeFissionability(Universe* univ) {
 
 
 /**
+ * @brief Get the material, cell or FSR IDs on a 2D spatial grid.
+ * @details This is a helper method for the openmoc.plotter module.
+ *          This method may also be called by the user in Python if needed.
+ *           A user must initialize NumPy arrays with the x and y grid
+ *          coordinates input to this function. This function then fills
+ *          a NumPy array with the domain IDs for each coordinate. An example
+ *          of how this function might be called in Python is as follows:
+ *
+ * @code
+ *          grid_x = numpy.arange(-2., +2., 100)
+ *          grid_y = numpy.arange(-2., +2., 100)
+ *          num_xy = 100**2
+ *          zcoord = 20.
+ *          domain_type = 'material'
+ *          domain_ids = geometry.getSpatialDataOnGrid(
+ *              grid_x, grid_y, num_xy, zcoord, domain_type)
+ * @endcode
+ *
+ * @param grid_x a NumPy array of the x-coordinates
+ * @param num_x the number of x-coordinates in the grid
+ * @param grid_y a NumPy array of the y-coordinates
+ * @param num_y the number of y-coordinates in the grid
+ * @param domains a NumPy array of the domain IDs
+ * @param num_xy the number of coordinates in the 2D grid
+ * @param zcoord the z-coordinate to use to find the domain IDs
+ * @param domain_type the type of domain ('fsr', 'material', 'cell')
+ */
+void Geometry::getSpatialDataOnGrid(double* grid_x, int num_x,
+				    double* grid_y, int num_y,
+				    int* domains, int num_xy,
+				    double zcoord, const char* domain_type) {
```

Comment:
If you use std::vector no typemap is needed. You just need to include
std_vector.i  (which we already do) and it will just work.

On Mon, Apr 4, 2016, 9:03 PM Will Boyd <notifications@github.com> wrote:

> In src/Geometry.cpp
> <https://github.com/mit-crpg/OpenMOC/pull/293#discussion_r58473658>:
>
> > + *              grid_x, grid_y, num_xy, zcoord, domain_type)
> > + * @endcode
> > + *
> > + * @param grid_x a NumPy array of the x-coordinates
> > + * @param num_x the number of x-coordinates in the grid
> > + * @param grid_y a NumPy array of the y-coordinates
> > + * @param num_y the number of y-coordinates in the grid
> > + * @param domains a NumPy array of the domain IDs
> > + * @param num_xy the number of coordinates in the 2D grid
> > + * @param zcoord the z-coordinate to use to find the domain IDs
> > + * @param domain_type the type of domain ('fsr', 'material', 'cell')
> > + */
> > +void Geometry::getSpatialDataOnGrid(double* grid_x, int num_x,
> > +				    double* grid_y, int num_y,
> > +				    int* domains, int num_xy,
> > +				    double zcoord, const char* domain_type) {
>
> It would be great to minimize this code since I do recognize this is quite
> clunky. I'd just rather not go hunting on another SWIG expedition to learn
> how to use std::vector. But if we have some examples in place already
> then I'll build upon them.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/OpenMOC/pull/293/files/e705c7159561da3597643f2d7a18d5ce67bbf8a4#r58473658>
>